pkg/manager/client.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manager
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// DefaultNewClient creates the default caching client
```

Comment:
This is still super inconvenient to use, can we just make the `WithUncached` an opt in the managers `Options` struct?

Reply:
What's inconvenient to use?

Reply:
Well, in order to set this, the whole call will look something like:
```
mgr, err := manager.New(cfg, manager.Optionss{
   NewClientFunc: manager.NewClientFuncBuilder().WithUncached(&SomeType{}),
}
```

Something like:
```
mgr, err := manager.New(cfg, manager.Options{
    DisableCacheFor: []runtime.Object{&SomeType{}},
}
```

is IMHO both easier to discover and easier to use. WDYT?

Reply:
I see, we could do both maybe? The only thing that throws me off with the second one is that the cache bypass is technically used with the client, not with a manager. What would happen for example if one sets `DisableCacheFor` list, and supply a `NewClientFunc` that doesn't support it?

Reply:
> The only thing that throws me off with the second one is that the cache bypass is technically used with the client, not with a manager.

Maybe call it `DisableClientCacheFor`?

> What would happen for example if one sets DisableCacheFor list, and supply a NewClientFunc that doesn't support it?

Well, we can add a comment to mention that, but generally I would expect ppl that set a NewClientFunc to realize that it will overwrite all client-related settings. AFAIK we also have a `DryRunClient` option that would get invalidated just the same.

Reply:
Oh! We can change the NewClientFunc to be the builder interface, which defaults to the internal builder, and it can call the methods it knows

Reply:
Does that sound good to you @alvaroaleman? I might also backport it in a non-breaking way

Reply:
But what does that change, we would still have to pass a `NewClientFunc` rather than just configuring the option we care about? IMHO this is really hard to discover

Reply:
@alvaroaleman Updated with what I was thinking, it's a middle ground between both, if you still don't like it happy to just remove the builder